models/workday_history/workday__worker_position_org_daily_history.sql

```diff
@@ -0,0 +1,92 @@
+-- depends_on: {{ ref('stg_workday__worker_position_organization_base') }}
+{{ config(enabled=var('employee_history_enabled', False)) }}
+
+{% if execute %} 
+    {% set first_last_date_query %}
+    with min_max_values as (
+        select 
+            min(_fivetran_start) as min_start,
+            max(_fivetran_start) as max_start 
+        from {{ ref('stg_workday__worker_position_organization_base') }}
+    )
+
+    select 
+        min_start,
+        case when max_start >= {{ dbt.current_timestamp() }}
+            then max_start
+            else {{ dbt.date_trunc('day', dbt.current_timestamp()) }} 
+        end as max_date
+    from min_max_values
+
+    {% endset %}
+
+    {% set start_date = run_query(first_last_date_query).columns[0][0]|string %}
+    {% set last_date = run_query(first_last_date_query).columns[1][0]|string %}
+
+{# If only compiling, creates range going back 1 year #}
+{% else %} 
+    {% set start_date = dbt.dateadd("year", "-2", "current_date") %} -- Arbitrarily picked. Choose a more appropriate default if necessary.
+    {% set last_date = dbt.dateadd("year", "-1", "current_date") %}
```

Comment:
Same suggestion from previous model for consistency and clarity for ourselves when we come back to this in the future.

```suggestion
{% else %} 
    {% set start_date = dbt.dateadd("year", "-1", "current_date") %} -- One year in the past for first date
    {% set last_date = dbt.dateadd("day", "-1", "current_date") %} -- Yesterday as last date
```

Reply:
Committed.